<a href="https://colab.research.google.com/github/mehta-a/100-days-of-code/blob/master/TensorFlow_Hello_World.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from __future__ import absolute_import, division, print_function, unicode_literals

!pip install tensorflow==2.0.0-beta1

import tensorflow as tf

     |████████████████████████████████| 87.9MB 1.2MB/s 
     |████████████████████████████████| 3.1MB 44.6MB/s 
     |████████████████████████████████| 501kB 46.8MB/s 
  Found existing installation: tensorflow 1.14.0
    Uninstalling tensorflow-1.14.0:
      Successfully uninstalled tensorflow-1.14.0


Loading MNIST Dataset



In [4]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0 # normalizing datasets

11493376/11490434 [==============================] - 0s 0us/step


Building tf.keras.sequential model by stacking layers, choose an optimizer and liss function for training:

In [0]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28,28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation='softmax')
])
model.compile(optimizer='adam',
             loss='sparse_categorical_crossentropy',
             metrics=['accuracy'])

Train and evaluate the model:

In [13]:
len(x_train), len(x_test)

(60000, 10000)

In [11]:
model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test, y_test)

W0827 10:50:41.437525 140411188946816 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 5s 89us/sample - loss: 0.2907 - accuracy: 0.9153
Epoch 2/5
60000/60000 [==============================] - 5s 82us/sample - loss: 0.1428 - accuracy: 0.9579
Epoch 3/5
60000/60000 [==============================] - 5s 82us/sample - loss: 0.1080 - accuracy: 0.9668
Epoch 4/5
60000/60000 [==============================] - 5s 82us/sample - loss: 0.0876 - accuracy: 0.9728
Epoch 5/5
10000/10000 [==============================] - 0s 38us/sample - loss: 0.0761 - accuracy: 0.9777


[0.07607756302407943, 0.9777]

Loss and accuracy in train set is very similar to test set. We were able to achieve close to 98% accuracy here.